### PR TITLE
Allow "[profile default]" in .aws/config

### PR DESCRIPTION
**Pull request by c42f**
The awscli allows either a section `[default]` or `[profile default]` in
the aws config file.

I had `[profile default]` in my config and was surprised when this worked with awscli but not with AWS.jl.

### CODE REVIEW

**Comment by omus** [comment]
Seems reasonable. Needs a test though

**Comment by c42f** [comment]
Test added.

File: test/AWSConfig.jl [hunk 1]
```diff
@@ -19,4 +19,18 @@
...
+            for default_section_str in allowed_default_sections
+                config = """
+                [$default_section_str]
+                region = xx-yy-1
+                """
+                write(config_path, config)
+                @test aws_get_region(; profile="default", config=config_path) == "xx-yy-1"
+            end
+        end
+    end
```
Comment (mattBrzezinski) [inline test/AWSConfig.jl:35]: I believe this is the proper syntax.

```suggestion
    @testset "$(profile) section names" for profile in ["default", "profile_default"]
        mktemp() do config_path, _
                config = """
                [$(profile)]
                region = xx-yy-1
                """
                write(config_path, config)
                @test aws_get_region(; profile="default", config=config_path) == "xx-yy-1"
            end
        end
    end
```

Comment (mattBrzezinski) [reply test/AWSConfig.jl:35]: It's nbd, I just find different cases in a `@testset` are easier to debug.

**Pull request merged**